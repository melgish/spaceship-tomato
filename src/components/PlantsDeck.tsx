import type { HTMLAttributes } from 'react'
import type { Plants, Plant } from '@/actions/findPlants'
import PlantCard from '@/components/PlantCard'

type ElementProps = Omit<HTMLAttributes<Element>, 'children'>
type Props = Readonly<ElementProps & { rows?: Plants }>

export default function PlantsDeck({ rows = [], ...props }: Props) {
  return (
    <div {...props}>
      {rows.map((row: Plant) => (
        <PlantCard key={row.id} {...row} />
      ))}
    </div>
  )
}
