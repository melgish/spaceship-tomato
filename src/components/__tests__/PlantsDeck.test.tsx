import { describe, expect, it, vi } from 'vitest'
import { render } from '@testing-library/react'
import PlantsDeck from '@/components/PlantsDeck'

vi.mock('@/components/PlantCard', () => ({
  default: ({ id }: { id: number }) => <div>{id}</div>,
}))

const data = [
  { id: 1, name: 'Plant 1' },
  { id: 2, name: 'Plant 2' },
] as unknown as Parameters<typeof PlantsDeck>[0]['rows']

describe('when there are no readings', () => {
  it('should render an empty div', () => {
    const { container } = render(<PlantsDeck />)

    expect(container).toMatchSnapshot()
  })
})

describe('when there are readings', () => {
  it('should render a card for each reading', () => {
    const { container } = render(<PlantsDeck className="test" rows={data} />)

    expect(container).toMatchSnapshot()
  })
})
