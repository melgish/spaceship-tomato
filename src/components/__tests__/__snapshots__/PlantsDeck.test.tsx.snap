// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when there are no readings > should render an empty div 1`] = `
<div>
  <div />
</div>
`;

exports[`when there are readings > should render a card for each reading 1`] = `
<div>
  <div
    class="test"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</div>
`;
